### PR TITLE
libobs: Add source callback create2

### PR DESCRIPTION
**Pull request by kkartaltepe**
Add an alternate create function for sources that will includes type
data for the source.

### CODE REVIEW

**Comment by kkartaltepe** [comment]
If you want to check out the module that uses this you can see it at https://github.com/kkartaltepe/obs-lua its still a bit rough around the edges.

File: libobs/obs-source.c [hunk 1]
```diff
@@ -332,9 +332,14 @@ static obs_source_t *obs_source_create_internal(const char *id,
 
 	/* allow the source to be created even if creation fails so that the
 	 * user's data doesn't become lost */
-	if (info)
-		source->context.data = info->create(source->context.settings,
-				source);
+	if (info) {
+		if(info->create2) /* alow plugins to use type data during creation */
```
Comment (RytoEX) [inline libobs/obs-source.c]: Spelling nitpick:  "alow" should be "allow"

File: libobs/obs-source.h [hunk 2]
```diff
@@ -423,6 +423,16 @@ struct obs_source_info {
 	void (*enum_all_sources)(void *data,
 			obs_source_enum_proc_t enum_callback,
 			void *param);
+
+	/**
+	 * Creates the source data for the source with type data.
+	 *
+	 * @param  settings  Settings to initialize the source with
+	 * @param  source    Source that this data is assoicated with
```
Comment (RytoEX) [inline libobs/obs-source.h]: Spelling nitpick:  "assoicated" should be "associated"

**Comment by reboot** [comment]
I don't see any reason why you need this. You can should be able to use obs_source_get_type_data() on the source that the create function gets as the second parameter to get the type_data from the obs_source_info.

**Comment by kkartaltepe** [comment]
Well there we go, that was the link I thought I was missing when I asked in IRC. I'm not sure how I missed it but thanks!

**Pull request closed without merging**